main.tf

```diff
@@ -1 +1,609 @@
-# Your Terraform code goes here :)
+
+data "aws_region" "current" {}
+
+// Todo: replace with iam role module
+resource "random_string" "external_id" {
+  length           = 256
+  override_special = "=,.@:/-"
+}
+
+// TF provider agentless scan resource
+
+resource "lacework_integration_aws_agentless_scanning" "lacework_cloud_account" {
+  name                      = var.cloud_integration_name
+  scan_frequency            = var.scan_frequency
+  query_text                = var.query_text
+  scan_containers           = var.scan_containers
+  scan_host_vulnerabilities = var.scan_host_vulnerabilities
+}
+
+// Global
+// SecretsManagers
+resource "aws_secretsmanager_secret" "agentless_scan_secret" {
+  name = "${var.resource_name_prefix}-secret-${var.resource_name_suffix}"
+}
+
+resource "aws_secretsmanager_secret_version" "agentless_scan_secret_version" {
+  secret_id     = aws_secretsmanager_secret.agentless_scan_secret.id
+  secret_string = <<EOF
+   {
+    "account": "${var.lacework_aws_account_id}",
+    "token": "${lacework_integration_aws_agentless_scanning.lacework_cloud_account.server_token}"
+   }
+EOF
+}
+
+// AWS::IAM::ServiceLinkedRole
+resource "aws_iam_service_linked_role" "agentless_scan_linked_role" {
+  aws_service_name = "ecs.amazonaws.com"
+  description      = "Role to enable Amazon ECS to manage your cluster."
+}
```

Comment:
In CloudFormation we had an input question (yes/no) where we asked people if they want our template to install the AWS ServiceLinkedRole.

I think we can skip this in Terraform as this role is a bit unique. Once it's added to an account it cannot be removed. We can provide instructions about how to set it up for the first time for now.

Reply:
👍  I've added a variable to determine whether this will be created or not. Defaulting to false.